extensions/pkg/controller/status.go

```diff
@@ -69,7 +69,7 @@ type StatusUpdater interface {
 	// Error updates the last operation of an extension resource when an operation was erroneous.
 	Error(context.Context, extensionsv1alpha1.Object, error, gardencorev1beta1.LastOperationType, string) error
 	// Success updates the last operation of an extension resource when an operation was successful.
-	Success(context.Context, extensionsv1alpha1.Object, gardencorev1beta1.LastOperationType, string) error
+	Success(context.Context, extensionsv1alpha1.Object, gardencorev1beta1.LastOperationType, string, func() error) error
```

Comment:
While the fix do work, I am wondering whether we should implement it this way.
This method is supposed to update only the last operation field with the provide message. 
Can we rather send dedicated update request to add the secret and units to the OSC status instead of re-using the last operation update method?

Reply:
Mmh, I have no strong opinion here.
- I liked this implementation because then only one patch request is sent and not two. 
- On the other side the osc is the only user of this mutate parameter and you are right - the `Success` method seems to do more than just one thing now. I will try out your suggestion.

Reply:
I tested it and the osc status got updated successfully.